boot/initramfs20dirs.go

```diff
@@ -97,7 +98,13 @@ func setInitramfsDirVars(rootdir string) {
 	InitramfsUbuntuBootDir = filepath.Join(InitramfsRunMntDir, "ubuntu-boot")
 	InitramfsUbuntuSeedDir = filepath.Join(InitramfsRunMntDir, "ubuntu-seed")
 	InitramfsUbuntuSaveDir = filepath.Join(InitramfsRunMntDir, "ubuntu-save")
-	InstallHostWritableDir = filepath.Join(InitramfsRunMntDir, "ubuntu-data", "system-data")
+	// On classic+modes systems the writable dir is directly under
+	// "ubuntu-data"
+	if release.OnClassic {
```

Comment:
this won't work for an install mode that itself is core-like. In theory this and dirs that derive from it need to become functions that take a boot.Device.

Do we need to change the InitramfsHostWritableDir too?

What about InitramfsWritableDir ?

Reply:
Just to double check, what you would prefer is something like:
```go
iff --git a/boot/initramfs20dirs.go b/boot/initramfs20dirs.go
index 488340cf3a..4fa717a31e 100644
--- a/boot/initramfs20dirs.go
+++ b/boot/initramfs20dirs.go
@@ -59,11 +60,6 @@ var (
        // physical partition depending on what system mode the system is in.
        InitramfsWritableDir string
 
-       // InstallHostWritableDir is the location of the writable partition of the
-       // installed host during install mode. This should always be on a physical
-       // partition.
-       InstallHostWritableDir string
-
        // InstallHostFDEDataDir is the location of the FDE data during install mode.
        InstallHostFDEDataDir string
 
@@ -89,6 +85,21 @@ var (
        snapBootFlagsFile string
 )
 
+type Classicfier interface {
+       Classic() bool
+}
+
+// InstallHostWritableDir is the location of the writable partition of the
+// installed host during install mode. This should always be on a physical
+// partition.
+func InstallHostWritableDir(d Classicfier) string {
+       if d.Classic() {
+               return filepath.Join(InitramfsRunMntDir, "ubuntu-data")
+       } else {
+               return filepath.Join(InitramfsRunMntDir, "ubuntu-data", "system-data")
+       }
+}

```
for all three dirs? And all callers need to provide a snap.Device (even if it's just a trivial one)?

Reply:
something like this, you probably need at least Classic() and RunMode(),  could be called InitramfsDevice